docs/scos/dev/front-end-development/202307.0/oryx/technology.md

```diff
@@ -0,0 +1,77 @@
+---
+title: "Oryx: Technology"
+description: The technologies Oryx is built on
+template: concept-topic-template
+last_updated: Oct 23, 2023
+---
+
+Oryx is a frontend framework designed to help developers build web applications using modern technologies and leveraging [Spryker API](https://docs.spryker.com/docs/scos/dev/glue-api-guides/{{page.version}}/decoupled-glue-api.html).
+
+Oryx leverages standard technology, frameworks and tools provided by the web platform and the community wherever possible.  It emphasizes utilizing the capabilities of the web and minimizing the reliance on build tools. During development, Oryx primarily employs JavaScript (JS) but aims to reduce the need for executing JS in the browser. It prioritizes HTML and CSS for delivering a fast-loading experience across various screen sizes.
+
+Furthermore, Oryx is framework-agnostic and can seamlessly integrate into various web frameworks like React and Vue.js. This versatility is achieved using a combination of web components for the user interface and vanilla JS for handling business logic. This approach offers the flexibility of using the entire library with its components or to selectively employing specific layers, like the integration logic.
+
+In terms of styling and layout rendering, Oryx adopts a CSS-first approach, considering CSS as the primary method. It recognizes the capabilities of CSS in handling styling and layout tasks without requiring additional JS logic. Oryx tries to avoid loading JS on the client side, loading snippets of JS rather than a full JS framework. This approach minimizes the overall JS payload.
+
+The following technologies are employed in the development of Oryx:
+- [HTML](#html)
+- [CSS](#css)
+- [TypeScript](#typescript)
+- [Web Components](#web-components)
+- [Lit](#lit)
+- [RxJS](#rxjs)
+- [Vite](#vite)
+- [Storybook](#storybook)
+- [Web hosting](#web-hosting)
+
+
+## HTML
+
+HTML is a markup language used to create web pages and applications. In Oryx, HTML is used to structure the content and layout of the application. It provides a semantic structure to the page, making it easier to understand and maintain. Also, Oryx prioritizes accessibility principles ensuring access to information and functionality for all users.
+
+## CSS
+
+CSS is a style sheet language used to describe the visual presentation of a document written in HTML.
+
+Oryx uses standard CSS syntax and does not use CSS preprocessors because Oryx does not use a global CSS utility library; instead, we have isolated styles per component and because of that we do not want to load global css utils in each component.
```

Comment:
Do we need to talk about not using preprocessors or the library?

Reply:
I'd say it's nice to have, but not necessary.